src/components/OptimizedImage.js

```diff
@@ -42,24 +44,76 @@ export default function OptimizedImage(props) {
     }
   }
 
-  return (
-    <>
-      <img
-        src={cdnUrl || src}
-        alt=".."
-        srcSet={srcSet}
-        sizes={sizes}
-        loading="lazy"
+  const handleImageError = (e) => (e.target.src = null);
+
+  const { pathname } = useLocation();
+  const checkPathName = (path) => pathname === path;
```

Comment:
This is a bit like a circular dependency in disguise.
`OptimizedImage` should not need to know about paths, and using the path to determine how to style the image is likely to cause problems when either the paths change or someone changes or adds image on one of these pages.

Instead, allow the caller to set the style it needs using props. A couple of different options come to mind:
1. Create a set of _modes_ or _roles_ that use pre-defined styles and allow the caller to specify which to use as a prop.
2. Pass these usage-specific style override attributes as props to `OptimizedImage` (e.g., `imgStyle`, `alertStyle`, `alertTitleStyle`).

Reply:
I understand your concerns. I'll see what I can do and maybe encapsulate the logic in a new ImageError component.